app/assets/javascripts/oxalis/view/action_bar_view.js

```diff
@@ -82,31 +84,30 @@ class ActionBarView extends React.PureComponent<Props, State> {
     }
   };
 
+  createTracing = async () => {
+    const type = "hybrid";
+    const annotation = await createExplorational(this.props.dataset, type, true);
+    trackAction(`Create ${type} tracing (from view mode)`);
+    location.href = `${location.origin}/annotations/${annotation.typ}/${annotation.id}${
+      location.hash
+    }`;
+  };
+
   renderStartTracingButton(): React.Node {
-    const createTracing = async () => {
-      const type = "hybrid";
-      const annotation = await createExplorational(this.props.dataset, type, true);
-      trackAction(`Create ${type} tracing (from view mode)`);
-      location.href = `${location.origin}/annotations/${annotation.typ}/${annotation.id}`;
+    const needsAuthentication = this.props.activeUser == null;
+
+    const handleCreateTracing = async () => {
+      if (needsAuthentication) {
+        this.setState({ isAuthenticationModalVisible: true });
```

Comment:
I'm not sure whether we want to get deeper into the `renderIndependently` pattern, but I think this could be useful here. Something like:

```js
const handleCreateTracing = async () => {
   const needsAuthentication = ....
   if (needsAuthentication) {
     const authenticated = await authenticateViaUI();
     if (!authenticated) {
      return;
     }
   }
   this.createTracing();
}

async function authenticateViaUI() {
  await renderIndependently(destroy =>
    <AuthenticationModal
      onLoggedIn={() => {
        destroy(true); // the destroy function would need to be adapted so that it's parameter is the value of the resolved promise
      }}
      onCancel={() => destroy(false)}
    />
  );
}
```

That way, we wouldn't have the additional `isAuthenticationModalVisible` state in this view (and react node in the render method). Also, the `createTracing` call is not in two different places. 

However, the current way is also fine. Just food for thought, since this scenario is likely to re-occur in the future.

Reply:
Actually, this was my first approach for exactly the same reasons you describe here! But, while developing I noticed that this would also have some drawbacks, the major one being that one couldn't use any of the `withRouter` functionality, because the component is being rendered without the correct Context Provider (not within a <Router .../>). I suppose rendering components that use `mapStateToProps` wouldn't work either :/
This led me to conclude that we should only use this pattern for small components with very encapsulated functionality and that the `isAuthenticationModalVisible` approach is actually cleaner. However, I'm open to discuss this with you :)
For now, I'd leave it as is.